brave-spancollector-scribe/src/test/java/com/github/kristofa/brave/scribe/CountingScribeSpanCollectorNotificationsHandler.java

```diff
@@ -0,0 +1,53 @@
+package com.github.kristofa.brave.scribe;
+
+import com.twitter.zipkin.gen.Span;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+class CountingScribeSpanCollectorNotificationsHandler implements ScribeSpanCollectorNotificationsHandler {
+
+    private final AtomicInteger spansRejected = new AtomicInteger();
+    private final AtomicInteger spansSentEvents = new AtomicInteger();
+    private final AtomicInteger sendingFailures = new AtomicInteger();
+    private volatile int lastQueueSize;
```

Comment:
We should make a benchmark project, so stuff like this is of more obvious impact. Wherever we try to track state, we'll likely inherit a performance hit. In this case, it is flushing to main memory on updates to these fields.